src/tutorials.ts

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) LSST DM/SQuaRE
+// Distributed under the terms of the MIT License.
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+
+import { PageConfig } from '@jupyterlab/coreutils';
+
+import { IDocumentManager } from '@jupyterlab/docmanager';
+
+import { ServerConnection } from '@jupyterlab/services';
+
+import { Menu } from '@lumino/widgets';
+
+//import { ServiceManager, ServerConnection } from '@jupyterlab/services';
+import { IMainMenu } from '@jupyterlab/mainmenu';
+
+import * as token from './tokens';
+
+enum Actions {
+  COPY = 'copy',
+  FETCH = 'fetch'
+}
+
+enum Dispositions {
+  PROMPT = 'prompt',
+  OVERWRITE = 'overwrite',
+  ABORT = 'abort'
+}
+
+interface ITutorialsEntryResponse {
+  menu_name: string;
+  action: Actions;
+  disposition: Dispositions;
+  parent: string;
+  src: string;
+  dest: string;
+}
+
+class TutorialsEntry implements ITutorialsEntryResponse {
+  menu_name: string;
+  action: Actions;
+  disposition: Dispositions;
+  parent: string;
+  src: string;
+  dest: string;
+
+  constructor(inp: ITutorialsEntryResponse) {
+    (this.menu_name = inp.menu_name),
+      (this.action = inp.action),
+      (this.disposition = inp.disposition),
+      (this.parent = inp.parent),
+      (this.src = inp.src),
+      (this.dest = inp.dest);
+  }
+}
+
+interface ITutorialsHierarchyResponse {
+  entries: { [name: string]: ITutorialsEntryResponse } | null;
+  subhierarchies: { [name: string]: ITutorialsHierarchyResponse } | null;
+}
+
+class TutorialsHierarchy implements ITutorialsHierarchyResponse {
+  entries: { [name: string]: TutorialsEntry } | null = null;
+  subhierarchies: { [name: string]: TutorialsHierarchy } | null = null;
+
+  constructor(inp: ITutorialsHierarchyResponse, name: string | null = null) {
+    if (name === null) {
+      name = '<unnamed>';
+    }
+    console.log(`Building hierarchy ${name}`);
```

Comment:
Also just noticed your message that logging would be removed so you may want to skip this, although perhaps you may still want to leave some logging in there that you can turn it off/on via an env var? 

May be useful in case things break and you won't have to add logging again from the start.

Reply:
Yeah, sort of.  But I don't have process.env in a useful fashion where the UI is running.  But that's why I have the environment extension...but I need to refactor the loading so I can load the environment before initializing the other extensions, and then figure out a way to stash it so they can get to it.

That feels like the big tech-debt refactor I'm putting off, though.  For the time being I'll probably get it debugged and then throw away a bunch of the console.log statements.